### DISCUSSION TITLE
Can't see properties

### DISCUSSION THREAD

**Discussion by quentinproust** [structurizr/lite#174] [Q&A] [answered]
Hello, I'm testing C4 diagrams. I tested Plantuml support and now Structurizr.

In plantuml, I can define the following diagram : 

```
@startuml
!include <C4/C4_Deployment>

AddProperty("Zone", "Secure")
System_Boundary(sys, "System") {
    AddProperty("dns", "api.somewhere.com")
    Container(api, "Api", "Super Api")
    
    Container(app, "App", "system core app")
}

AddProperty("Port", "8080")
Rel(api, app, "Uses")

@enduml
```

which gives this : 

![PP2n2W8n38RtFaNjT8AeWuE3Y7e-W2WkBd9S0nxSDILfaIA-ksblKhCb_vV_9zbnQ2Z4eLUJphLzjA0tzMfHh-v7y3sc0LpOAhMtzaJeWKAgp0qTc9as5sWZWPcgI-80m_s0qTc6KiM9irocPgfVIajzMLZ7sQ7ntPnnWES32EOj2btO6breEWTK2I7StdSbCCgwpis8_R6-i3wpND](https://github.com/user-attachments/assets/3ff2cc7d-11a0-4dbe-9b48-795612df053d)

When I tried to do the same thing in Structurizr, I can't see the properties and I'm not sure if I missed something or if it's not displayed.

The structurizr code : 

```
workspace {

    !identifiers hierarchical

    model {
        x = softwareSystem "X" {
            api = container "Api" "Super Api" {
                properties {
                    dns api.somewhere.com
                }
            }
            app = container "App" "system core app"
            
            api -> app "Uses" {
                properties {
                    port 8080
                }
            }
        }
    }

    views {
        container x "Container-X" {
            include *
            autolayout lr
        }
    }
}
```

I got this image in Structurizr Lite (I checked also in Structure On Premise Live Editor and using Kroki.io) : 

![Sans titre](https://github.com/user-attachments/assets/0060130f-d999-417b-9895-ffa091fcf222)

I checked the documentation, issues and discussions to see if I could find something but found nothing.

Am I missing something or is this feature (showing properties) not supported ?

If it's not supported, do you have best practices for this type of documentation ?

To be completely honest, my use case is a mkdocs static site where I use Kroki.io to export different diagrams to be included in the documentation 😃

**Comment by simonbrowndotje** [comment]
Properties are displayed in Structurizr Lite via tooltips when the mouse is hovered over an element/relationship ... press the `t` key when viewing a diagram to toggle tooltips on/off.

**Comment by quentinproust** [comment]
Thank you for the answer, I found the button to toggle the tooltips when I pressed `t` 😄 

If I understand correctly, it means won't be displayed when I export the diagram too.
I tried it and it wasn't visible, juste asking to be absolutely sure I didn't miss a simple button again 🙂

If that's the case, can you tell if it's possible in another "version" of structuriz (on premise, cli, ...) ?

**Reply by simonbrowndotje** [reply, marked as answer]
Structurizr Lite/on-premises/cloud all use the same UI, so properties are only visible via tooltips. If you need properties on the diagram, you would need to use the [C4PlantUMLExporter](https://docs.structurizr.com/export/plantuml#c4plantumlexporter) that is bundled into the Structurizr CLI, in conjunction with `c4plantuml.elementProperties true` (see linked docs; example from the demo page below).

<img width="1728" alt="Screenshot 2024-12-09 at 09 04 40" src="https://github.com/user-attachments/assets/9aaf26c6-e164-4062-a8a9-6da6bbd6efb7">
